src/test/ui/consts/const_in_pattern/issue-65466.rs

```diff
@@ -0,0 +1,19 @@
+#![deny(indirect_structural_match)]
+
+#[derive(PartialEq, Eq)]
+enum O<T> {
+    Some(*const T), // Can also use PhantomData<T>
```

Comment:
this is as much as note to myself as to @ecstatic-morse : Do we need a variant of this test that uses `PhantomData<T>`? (Maybe just as another struct definition in this same file and a corresponding additional `const` and `match` below?)